### PR TITLE
[PBW-3956] thumbnail seeking (please do not merge, for review only)

### PR DESCRIPTION
**Pull request by ksunika**
skin config change to go with it:
https://github.com/ooyala/skin-config/pull/138

### CODE REVIEW

File: js/controller.js [hunk 1]
```diff
@@ -314,9 +316,479 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
...
+         300,
+         310,
+         320,
+         330
+      ],
+"available_widths":[
+         120,
+         240
+      ],
+      "thumbnails":{
```
Comment (dchauOoyala) [inline js/controller.js]: This may have already been addressed during yesterday's demo, but eventually this data will not be hard-coded, correct?

Comment (ksunika) [inline js/controller.js]: @dchauOoyala, yes, definitely. The real data will be coming with the appropriate event.

File: js/components/scrubberBar.js [hunk 2]
```diff
@@ -206,11 +234,46 @@ var ScrubberBar = React.createClass({
...
+        (<div className="oo-scrubber-thumbnail-container">
+          <Thumbnail
+            thumbnails={this.props.controller.state.thumbnails}
+            hoverPosition={hoverPosition}
+            duration={this.props.duration}
+            hoverTime={hoverTime > 0 ? hoverTime : 0}
+            scrubberBarWidth={this.state.scrubberBarWidth}/>
+        </div>);
+    }
+    else var thumbnailContainer = null;
```
Comment (chibzu) [inline js/components/scrubberBar.js]: I would just move this to above the if statement without the else, let it be the default. Then you don't need this awkward single line else at the end.

**Comment by chibzu** [comment]
This is awesome, looks great and works like a charm. Code and tests look great also. Only found one thing so far, on smaller players the thumbnail seems to appear left of where your cursor is currently hovering. Is this by design or a side effect?

**Comment by ksunika** [comment]
@chibzu, thank you! The small player thumbnails being not centered is definitely not by design, I will fix that.

File: js/controller.js [hunk 3]
```diff
@@ -122,6 +123,7 @@ OO.plugin("Html5Skin", function (OO, _, $, W) {
       this.mb.subscribe(OO.EVENTS.DESTROY, 'customerUi', _.bind(this.onPlayerDestroy, this));
       this.mb.subscribe(OO.EVENTS.EMBED_CODE_CHANGED, 'customerUi', _.bind(this.onEmbedCodeChanged, this));
       this.mb.subscribe(OO.EVENTS.CONTENT_TREE_FETCHED, 'customerUi', _.bind(this.onContentTreeFetched, this));
+      this.mb.subscribe(OO.EVENTS.STYLE_FETCHED, 'customerUi', _.bind(this.onStyleFetched, this));//xenia: to be replaced by a more appropriate event
```
Comment (aldigun) [inline js/controller.js]: This event listener has to be changed. I am proposing to use CMS-less new event: ASSET_UPDATED

File: js/components/thumbnail.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    return {
+      thumbnailWidth: 0
+    };
+  },
+
+  componentDidMount: function() {
+    this.setState({thumbnailWidth: ReactDOM.findDOMNode(this.refs.thumbnail).clientWidth});
+  },
+
+  // findThumbnail: function(hoverTime) { //ESPN assets only, core branch pbw-5132
```
Comment (aldigun) [inline js/components/thumbnail.js]: Do we need the commented out chunk here?

File: js/components/thumbnail.js [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+  //   return selectedThumbnail;
+  // },
+
+  findThumbnail: function(hoverTime) {
+    var thumbnails = this.props.thumbnails;
+    var timeSlices = thumbnails.data.available_time_slices;
+    var width = this.props.thumbnails.data.available_widths[0]; //choosing the lowest size
+    var selectedThumbnail = null;
+
+    var position = Math.floor((hoverTime/this.props.duration)* timeSlices.length);
```
Comment (aldigun) [inline js/components/thumbnail.js]: space before *

**Comment by aldigun** [comment]
@chibzu this merge has conflict.

**Comment by aldigun** [comment]
looks good to me.

**Pull request merged**